openquake/hazardlib/tests/gsim/zhao_2016_perg_test.py

```diff
@@ -0,0 +1,322 @@
+    dist_x = ctxs.rrup
+    mean_perg = mean_perg[0][0]
+    mean = mean[0][0]
+    pyplot.plot(dist_x, np.exp(mean_perg), 'r', label = 'PErg')
+    pyplot.plot(dist_x, np.exp(mean), 'b', label = 'Non-PErg')
+    pyplot.semilogy()
+    pyplot.ylabel('%s (g)' %(imt))
+    pyplot.xlabel('Joyner-Boore distance (km)')
+    pyplot.title('Test scenario for imt = %s and site azimuth = %s$^o$' %(
+        imt, azimuth))
+    pyplot.legend()
+    pyplot.show()
+    
+    return mean_perg, mean
+    
+        
+class TestZhao2016PErg(unittest.TestCase):
+    """
+    Test implementation of non-ergodic path modifications as described within
+    the Zhao et al. (2016) GMMs. 5 volcanic zone polygons including complex
+    shapes (i.e. ones which may be traversed multiple times by the same travel
+    path) are considered here. The test volcanic zone polygons are in 
+    openquake.hazardlib.tests.gsim.data.ZHAO16PERG. Multiple spectral ordinates
+    are also considered.
+    
+    The test scenarios below consider sites generated w.r.t. the same rupture,
+    but with different site azimuths, resulting in different travel path
+    configurations through these volcanic zone polygons. The expected distances
+    for many scenarios also not considered here were measured using QGIS, and
+    excellent matches were observed when compared against the values computed
+    using the ray tracing functions implemented here. These ray-tracing functions
+    are found within openquake.hazardlib.gsim.zhao_2016_volc_perg.volc_perg
+    
+    The tests below test the execution of the non-ergodic implementation
+    of ZhaoEtAlSSlabPErg, and demonstrate the difference in predicted
+    ground-motion if these path modifications are considered.
+    """
+    def setUp(self):
+        """
+        Create rupture context and setup non-ergodic and ergodic implementations
+        of Zhao et al. (2016) intra-slab GMM.
+        """
+        # Get rupture
+        lon = 108.08
+        lat = 28.328
+        dep = 10.0
+        msr = WC1994()
+        mag = 7.0
+        aratio = 2.0
+        strike = 270.0
+        dip = 30.0
+        rake = 90.0
+        trt = TRT.SUBDUCTION_INTERFACE
+        ztor = 0.0
+        self.rup = get_rupture(lon, lat, dep, msr, mag, aratio, strike, dip,
+                          rake, trt, ztor)
+        
+        # Get implementations of Zhao et al. (2016) intra-slab GMM
+        volc_arc_fname = DATA_FOLDER
+        self.gmm_perg = ZhaoEtAl2016SSlabPErg(volc_arc_fname)
+        self.gmm = ZhaoEtAl2016SSlab()
+        
+    def test01(self):
+        """
+        Test with azimuth of 90 degrees from rupture strike to sites collection
+        in straight line for SA(0.5) through test volcanic zone polygons.
+        """
+        # General inputs
+        imt = 'SA(0.5)'
+        
+        # Get sites from rupture
+        from_point = 'TC'
+        azimuth = 90 # relative to strike of slab
+        direction = 'positive'
+        hdist = 5000
```

Comment:
5000 km of distance?? Perhaps you meant 5000 meters?

Reply:
The distance needs to be 5000 km because we are testing the non-ergodic path effect corrections with the travel paths crossing through many large polygons representing volcanic regions. The distance needs to be this large to cross all of the (very large) zones in each test case.